server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -521,8 +522,22 @@ export class AgenticChatController implements ChatHandlers {
                 break
             }
 
-            // Process tool uses and update the request input for the next iteration
-            const toolResults = await this.#processToolUses(pendingToolUses, chatResultStream, session, tabId, token)
+            let toolResults: ToolResult[]
+            if (result.success) {
+                // Process tool uses and update the request input for the next iteration
+                toolResults = await this.#processToolUses(pendingToolUses, chatResultStream, session, tabId, token)
+            } else {
+                // Send an error card to UI?
+                toolResults = pendingToolUses.map(toolUse => ({
+                    toolUseId: toolUse.toolUseId,
+                    status: ToolResultStatus.ERROR,
+                    content: [{ text: result.error }],
+                }))
+                if (result.error.startsWith('ToolUse input is invalid JSON:')) {
+                    currentRequestInput.conversationState!.currentMessage!.userInputMessage!.content =
+                        "Your toolUse input isn't valid. Please check the syntax and make sure the input is complete. If the input is large, break it down into multiple tool uses with smaller input."
+                }
+            }
             currentRequestInput = this.#updateRequestInputWithToolResults(currentRequestInput, toolResults)
```

Comment:
Asking out of ignorance, would this method overwrite the currentRequestInput?

Reply:
it's already been used at this point for the current request, so this is really updating for the next request